src/imitation/data/rollout.py

```diff
@@ -538,8 +565,19 @@ def flatten_trajectories(
     Returns:
         The trajectories flattened into a single batch of Transitions.
     """
+
+    def all_of_type(key, desired_type):
+        return all(
+            isinstance(getattr(traj, key), desired_type) for traj in trajectories
+        )
+
+    assert all_of_type("obs", types.DictObs) or all_of_type("obs", np.ndarray)
+    assert all_of_type("acts", np.ndarray)
+
+    # mypy struggles without Any annotation here.
+    # The necessary constraints are enforced above.
```

Comment:
👍 this seems like right choice (trying to type check this all probably not worth the pain)